backend/api/routes/events.js

```diff
@@ -153,15 +199,37 @@ router.route("/rsvp").post(authenticateMiddleware, async (req, res) => {
 
     //found on supabase documentation interesting query "upsert" will add a new row for the rsvp if it doesnt exist but if user clicked like not going after
     //rsvping this will update it with that status. Pretty useful
-    const { error } = await supabaseClient.from("event_rsvps").upsert({
-      event_id,
-      user_id,
-      status,
-    }, {onConflict: "event_id, user_id"});
+    const { error } = await supabaseClient.from("event_rsvps").upsert(
+      {
+        event_id,
+        user_id,
+        status,
+      },
+      { onConflict: "event_id, user_id" }
+    );
 
     if (error) throw error;
 
     res.status(200).json({ success: true, message: "RSVP updated" });
+
+    const { data: event } = await supabaseClient
+      .from("events")
+      .select("creator_id, title")
+      .eq("id", event_id)
+      .single();
+
+    //dont notify if you yourself are rsvping to own event
+    if (event && event.creator_id !== user_id) {
+      createNotification(supabaseClient, {
```

Comment:
This could be a future improvement, but for these cases you can leverage classes and inheritance using the [factory design pattern](https://refactoring.guru/design-patterns/factory-method). So you create one class for each notification with things like message or title, then create a super class that will instantiate the child depending on the notification type. Is easier to maintain and understand!